website/addons/dataverse/model.py

```diff
@@ -88,22 +94,45 @@ class AddonDataverseNodeSettings(AddonNodeSettingsBase):
 
     dataverse_alias = fields.StringField()
     dataverse = fields.StringField()
-    study_hdl = fields.StringField()
-    study = fields.StringField()
+    dataset_doi = fields.StringField()
+    _dataset_id = fields.StringField()
+    dataset = fields.StringField()
+
+    # Legacy fields
+    study_hdl = fields.StringField()    # Now dataset_doi
+    study = fields.StringField()        # Now dataset
 
     user_settings = fields.ForeignField(
         'addondataverseusersettings', backref='authorized'
     )
 
     @property
-    def is_fully_configured(self):
-        return bool(self.has_auth and self.study_hdl is not None)
+    def dataset_id(self):
```

Comment:
The only time this work occurs is when `dataset_id` is accessed for the first time after we make the switch to 4.0. For all future projects, we can set it as we attach the dataset to the project. Normally, we could run a migration, but it is impossible to run this migration because of a simultaneous change in how Dataverse is handling authentication.

The only viable alternative I could think of would be to perform the check to see the dataset ID exists every time it is referenced. In that case though, it seems a lot safer to force that check through a property than it would be to make a note and hope that if `dataset_id` ends up being used other places (which it might as Dataverse adds in simpler "native" routes to replace the SWORD API), developers would remember to perform the check.